From 6d0617a7296af86996a7eddc07eca333b96fd15d Mon Sep 17 00:00:00 2001
Message-Id: <6d0617a7296af86996a7eddc07eca333b96fd15d.1428007496.git.feitong.yi@intel.com>
In-Reply-To: <96b4927b026b1cf56e9298f537d7ca517a5902b4.1428007496.git.feitong.yi@intel.com>
References: <96b4927b026b1cf56e9298f537d7ca517a5902b4.1428007496.git.feitong.yi@intel.com>
From: Namrta <namrta.salonie@intel.com>
Date: Fri, 20 Mar 2015 23:08:13 +0530
Subject: [PATCH 3/6] FOR_UPSTREAM [VPG]: drm/i915: Fix for potential NULL
 pointer dereference.

Ctx validity needs to be checked before de-referencing it.
Found by static analysis tool.

Issue: GMINL-7286
Change-Id: I35434e50cfe11d3af99bba529386adbc33bf36df
Signed-off-by: Namrta <namrta.salonie@intel.com>
---
 drivers/gpu/drm/i915/i915_gpu_error.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gpu_error.c b/drivers/gpu/drm/i915/i915_gpu_error.c
index 15c7c5a..c7f6d6f 100644
--- a/drivers/gpu/drm/i915/i915_gpu_error.c
+++ b/drivers/gpu/drm/i915/i915_gpu_error.c
@@ -978,7 +978,7 @@ static void i915_gem_record_rings(struct drm_device *dev,
 			 * for it to be useful (e.g. dump the context being
 			 * executed).
 			 */
-			if (request)
+			if (request && request->ctx)
 				rbuf = request->ctx->engine[ring->id].ringbuf;
 			else
 				rbuf = ring->default_context->engine[ring->id].ringbuf;
-- 
1.7.9.5

